src/sentry/tasks/assemble.py

```diff
@@ -311,14 +326,91 @@ def _create_artifact_bundle(
                     artifact_bundle=artifact_bundle,
                     source_file_type=source_file_type.value,
                 )
+        else:
+            raise AssembleArtifactsError(
+                "uploading a bundle without debug ids or release is prohibited"
+            )
+
+
+def get_with_manifest_fallback(value, manifest_key, manifest, throw=True):
+    if value is None:
+        manifest_value = manifest.get(manifest_key)
+        if manifest_value is None and throw:
```

Comment:
Hmm, what happens if `dist` is correctly loaded, but it was `null` in manifest, thus return `None` here? Will it throw? Or do we not store `null` in the manifest and it will barf because of the missing key? `dist` is optional in the end.

Reply:
That's a good observation. In my mind I assumed that None set is equal to unset.

Reply:
With @mitsuhiko we agreed that we want to give priority to the values in the request instead of the manifest.